var/spack/repos/builtin/packages/phist/package.py

```diff
@@ -0,0 +1,136 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+import os
+import fnmatch
+from spack import *
+
+
+class Phist(CMakePackage):
+    """ PHIST - a Pipelined, Hybrid-parallel Iterative Solver Toolkit.
+        
+PHIST provides implementations of and interfaces to block iterative solvers for sparse linear and eigenvalue problems.
+In contrast to other libraries we support multiple backends (e.g. Trilinos, PETSc and our own optimized kernels),
+and interfaces in multiple languages such as C, C++, Fortran 2003 and Python. PHIST has a clear focus on
+portability and hardware performance: in particular we support row-major storage of block vectors and using GPUs (via
+the ghost library or Trilinos/Tpetra).
+"""
+
+    homepage = "https://bitbucket.org/essex/phist/"
+    
+    # run expensive test suite (recommended oly if you have 12 physical cores, takes a few minutes)
+    run_tests  = False
+    smoke_test = True
+
+    version('develop',
+            git='https://bitbucket.org/essex/phist/phist.git', branch='devel')
+    version('master',
+            git='https://bitbucket.org/essex/phist/phist.git', branch='master')
+    version('1.4.1', '53ca2c2c000a36790e1626ed1eb642e3',
+            url='https://bitbucket.org/essex/phist/get/phist-1.4.1.tar.gz')
+
+    # note: there is no virtual package for lapacke (the C bindings for lapack),
+    # We look for a file like lapacke.h in the lapack prefix, if it is found we
+    # pass it to phist's cmake, otherwise we hope that phist finds it itself.
+    def lapacke_include_dir(self):
```

Comment:
do you mind moving this into blas/lapack packages? I think they should be responsible for this task, instead of every package using them.

Look at how query parameters are generated for `libs()` in [hdf5](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/hdf5/package.py#L107-L167), one can do the same i believe with `headers()`. 

So the idea is to have something like `spec['lapack:e'].headers` (`e` is query parameter) to get what you need.

I think there was a topic somewhere discussing lapacke, but i can't find it now.

Reply:
looks like we call the query param `c` for `libs()`, see [netlib-lapack](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/netlib-lapack/package.py#L104-L124). But AFAIK `headers()` are not implemented for lapacke.

Reply:
ah, thanks for the info....
I do not feel comfortable (being new to spack and not exactly a Python pro) introducing this into other packages, but it sounds like an excellent idea to me.

Reply:
> I do not feel comfortable (being new to spack and not exactly a Python pro) introducing this into other packages, but it sounds like an excellent idea to me.

ok, let me give it a quick spin, I am not a python pro either, btw 😄